### PR TITLE
Renames mergeSchema into the more accurate mergeResolvers

### LINKED ISSUE #752: Fix naming of mergeSchema
### Description
We have a `mergeSchema` function in apollos-core that doesn't actually merge schemas, it merges resolvers. We should rename the `mergeSchema` function `mergeResolvers`.

### Acceptance Criteria

* [ ] mergeSchema should be exported as mergeResolvers

### Other Resources

* Design mockup/prototype link
* Basecamp todo or link to "epic" on GitHub

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

`mergeSchema` naming was very confusing, should be `mergeResolver`

### What design trade-offs/decisions were made?

Opted to avoid creating an actual `mergeSchema` for now.

### How do I test this PR?

1. Run the tests ;) 

### What automated tests did you write?

Updated existing tets.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes #752
- [x] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by IsaacHardy** [comment]
Will the upgrade tool handle the name change within a churches persona API server?

**Comment by vinnyjth** [comment]
@IsaacHardy Nope. Because we are pre-1.0 I'm taking advantage of my opportunity to make breaking changes without needing a solution besides "help people upgrade" ;)

**Comment by vinnyjth** [comment]
@IsaacHardy I felt guilty so I added a `console.error`

File: packages/apollos-server-core/src/utils/index.js [hunk 1]
```diff
@@ -23,3 +23,7 @@ export const schemaMerge = (newResolver, { resolver = {} }) => {
   });
   return finalResolver;
 };
+
+export const schemaMerge = () => {
```
Comment (IsaacHardy) [inline packages/apollos-server-core/src/utils/index.js]: You're doing the Lord's work.

File: packages/apollos-server-core/src/utils/index.js [hunk 2]
```diff
@@ -23,3 +23,7 @@ export const schemaMerge = (newResolver, { resolver = {} }) => {
...
+  console.error('`schemaMerge` has been remove. Please use resolverMerge');
```
Comment (redreceipt) [inline packages/apollos-server-core/src/utils/index.js]:
```suggestion
  console.error('schemaMerge has been removed. Please use resolverMerge.');
```
